packages/react/src/themes/teams/components/Checkbox/checkboxStyles.ts

```diff
@@ -1,13 +1,107 @@
 import { ComponentSlotStylesInput, ICSSInJSStyle } from '../../../types'
-import { CheckboxProps, CheckboxState } from '../../../../components/Checkbox/Checkbox'
+import Checkbox, { CheckboxProps, CheckboxState } from '../../../../components/Checkbox/Checkbox'
 import { CheckboxVariables } from './checkboxVariables'
 import getBorderFocusStyles from '../../getBorderFocusStyles'
 
 const checkboxStyles: ComponentSlotStylesInput<CheckboxProps & CheckboxState, CheckboxVariables> = {
   root: ({ props: p, variables: v, theme: t }): ICSSInJSStyle => ({
+    color: v.checkboxTextColor,
+    padding: v.rootPadding,
+
     ...getBorderFocusStyles({
       siteVariables: t.siteVariables,
       isFromKeyboard: p.isFromKeyboard,
+      borderRadius: '3px', // shouldn't default be correct?
+    }),
+    ...(p.disabled && {
+      color: v.disabledColor,
+      cursor: 'default',
+    }),
+    ...(!p.disabled && {
+      ':hover': {
+        color: v.checkboxTextColorHover,
+        [`& .${Checkbox.slotClassNames.indicator}`]: {
+          ...(p.checked && {
+            background: v.checkboxCheckedBackgroundHover,
+          }),
+          ...(!p.checked && {
+            borderColor: v.checkboxBorderColorHover,
+            ...(p.toggle && {
+              color: v.checkboxCheckedBackgroundHover,
+            }),
+          }),
+        },
+      },
+    }),
+    ...(p.checked &&
+      !p.disabled && {
+        color: v.checkboxCheckedTextColor,
+      }),
+  }),
+
+  checkbox: ({ props: p, variables: v }): ICSSInJSStyle => ({
+    background: v.checkboxBackground,
+    borderColor: v.checkboxBorderColor,
+    borderStyle: v.checkboxBorderStyle,
+    borderRadius: v.checkboxBorderRadius,
+    borderWidth: v.checkboxBorderWidth,
+    color: v.checkboxIndicatorColor,
+    margin: v.checkboxMargin,
+    padding: v.checkboxPadding,
+
+    ...(p.checked && {
+      background: v.checkboxCheckedBackground,
+      borderColor: v.checkboxCheckedBorderColor,
+      color: v.checkboxCheckedIndicatorColor,
+    }),
+
+    ...(p.disabled && {
+      background: v.disabledCheckboxBackground,
+      borderColor: v.disabledCheckboxBorderColor,
+
+      ...(p.checked && {
+        color: v.disabledCheckboxCheckedIndicatorColor,
+        background: v.disabledCheckboxBackgroundChecked,
+      }),
+    }),
+  }),
+
+  label: ({ props: p, variables: v }): ICSSInJSStyle => ({
+    [p.labelPosition === 'start' ? ':after' : ':before']: {
+      content: '" "',
+      display: 'inline-block',
+      width: v.checkboxGap,
+    },
+  }),
```

Comment:
We don't need this section with `label` at all because it duplicates all styles from `base` theme